server/src/test/java/org/kaazing/gateway/server/util/version/DuplicateJarFinderTest.java

```diff
@@ -0,0 +1,114 @@
+            never(gatewayLogger).error(with(any(String.class)), with(any(Object.class)));
+        }});
+
+        try {
+            duplicateJarFinder.findDuplicateJars();
+        } catch (Exception e) {
+            fail(EXCEPTION_NOT_EXPECTED_MSG);
+        }
+
+    }
+    
+    @Test
+    public void testFindDuplicateJarsShouldNotThrowExceptionIfNoneKaazingProduct() throws IOException {
+        final ClassPathParser classPathParser = context.mock(ClassPathParser.class);
+        final Logger gatewayLogger = context.mock(Logger.class);
+        DuplicateJarFinder duplicateJarFinder = new DuplicateJarFinder(gatewayLogger);
+        duplicateJarFinder.setClassPathParser(classPathParser);
+
+        context.checking(new Expectations() {{
+            oneOf(classPathParser).getClassPathEntries();
+            will(returnValue(MOCK_CLASS_PATH_ENTRIES));
+            oneOf(classPathParser).getManifestAttributesFromClassPathEntry(MOCK_CLASS_PATH_ENTRIES[0]);
+            will(returnValue(getAttributesForNoneKaazingProduct()));
+            allowing(gatewayLogger).debug(with(any(String.class)), with(any(Object.class)), with(any(Object.class)));
+            never(gatewayLogger).error(with(any(String.class)), with(any(Object.class)));
+        }});
+
+        try {
+            duplicateJarFinder.findDuplicateJars();
+        } catch (Exception e) {
+            fail(EXCEPTION_NOT_EXPECTED_MSG);
+        }
+    }
+
+    @Test(expected=DuplicateJarsException.class)
+    public void testFindDuplicateJarsShouldThrowExceptionIfDuplicateKaazingProducts() throws IOException, DuplicateJarsException {
+        final ClassPathParser classPathParser = context.mock(ClassPathParser.class);
+        final Logger gatewayLogger = context.mock(Logger.class);
+        DuplicateJarFinder duplicateJarFinder = new DuplicateJarFinder(gatewayLogger);
+        duplicateJarFinder.setClassPathParser(classPathParser);
+
+        context.checking(new Expectations() {{
+            oneOf(classPathParser).getClassPathEntries();
+            will(returnValue(MOCK_CLASS_DUPLICATE_PATH_ENTRIES));
+            allowing(classPathParser).getManifestAttributesFromClassPathEntry(MOCK_CLASS_DUPLICATE_PATH_ENTRIES[0]);
+            will(returnValue(getAttributesForKaazingProduct()));
+            allowing(gatewayLogger).debug(with(any(String.class)), with(any(Object.class)), with(any(Object.class)));
+            allowing(gatewayLogger).error(with(any(String.class)), with(any(Object.class)));
+        }});
+
+        duplicateJarFinder.findDuplicateJars();
+
+        fail(EXCEPTION_EXPECTED_MSG);
+    }
+    
+    private Attributes getAttributesForKaazingProduct() {
+        Attributes attributes = new Attributes();
+        attributes.putValue("Implementation-Version", "1.0");
+        attributes.putValue("Artifact-Name", MOCK_JAR_FILE_NAME);
+        return attributes;
+    }
+
+    private Attributes getAttributesForNoneKaazingProduct() {
```

Comment:
getAttributesForKaazingProduct() and getAttributesForNoneKaazingProduct() seem to do the same thing